src/expire.c

```diff
@@ -343,16 +403,16 @@ void activeExpireCycle(int type) {
                          *             = avg_ttl +  (db->avg_ttl - avg_ttl) * pow(0.98, update_avg_ttl_times)
                          * Notice that update_avg_ttl_times is between 1 and 16, we use a constant table
                          * to accelerate the calculation of pow(0.98, update_avg_ttl_times).*/
-                        db->avg_ttl = avg_ttl + (db->avg_ttl - avg_ttl) * avg_ttl_factor[update_avg_ttl_times - 1];
+                        db->expiry[jobType].avg_ttl = avg_ttl + (db->expiry[jobType].avg_ttl - avg_ttl) * avg_ttl_factor[update_avg_ttl_times - 1];
                     }
                     update_avg_ttl_times = 0;
                     data.ttl_sum = 0;
                     data.ttl_samples = 0;
                 }
                 if ((iteration & 0xf) == 0) { /* check time limit every 16 iterations. */
                     elapsed = ustime() - start;
-                    if (elapsed > timelimit) {
-                        timelimit_exit = 1;
+                    if (elapsed > (long)endtime_us) {
```

Comment:
Bug.  Comparing an elapsed duration to an absolute end time.

```suggestion
                    if (getMonotonicUs() >= endtime_us) {
```

Reply:
We no longer use absolute end time.